### ISSUE TITLE
button repeat rate seems too high (Libretro)

### ISSUE THREAD

**Issue by jvook** [hrydgard/ppsspp#14631]
### What happens?
Not sure the best way to describe this issue but bare with me.  On the libretro core, when pressing a button it seems the "repeat rate" is too high.  Pressing down on d-pad during a menu might skip an entry or two, as if the button was pressed twice.  It requires a very quick tap in order to register just a single button press.

### What *should* happen?
Cursor in menus, etc. should move a single entry for each d-pad tap.

### What are you using?
Libretro
#### What hardware / device and operating system?
N/a
#### What graphics card (GPU) or mobile phone model?
N/a
#### What PPSSPP version (standalone/official), and did it work before?
No issues on PSP standalone AFAIK
#### Which game or games?

FF Tactics: button presses in menu will skip entries.  This can be partly mitigated by changing an in-game setting which controls cursor repeat rate, hence the name of this report.
Persona 2 US: button presses on difficulty selector when starting a new game will skip over medium to hard.  It's not just d-pad in this game, also when rotating the camera with R1/L1, it will skip rotation angles.  (However the pause menu does not skip entries -- this may have to do with the programming of the game but I think there is still present a fundamental input issue)
Note: because this is present in these two games I am filing this report and thinking that it is an issue with the libretro port.  If I should do more testing please advise.

#### Checklist

 - [n/a ] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
 - [x ] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
 - [ x] Try resetting settings or older versions and include if the issue is related.
 - [ n/a] Include logs or screenshots of issue.

**Comment by LunaMoo** [comment]
If in the real PPSSPP repeating in menus works, the problem lies within your third party UI and however it handles this function, which is not a PPSSPP issue. Closing.

**Comment by hrydgard** [comment]
Eh, it could be a problem with the code that lives in /libretro in PPSSPP. 

I don't mind having issues like this open, assuming there's an actual problem.

**Comment by bslenul** [comment]
I personally can't reproduce in FFT (I went in "Tutorial > Misc" menu to test, but maybe it happens in specific menus?).
In Persona 2 when selecting the difficulty however yes, it often scrolls 2 lines at once, but in my case it's not a Libretro core issue as it happens in standalone as well, no clue if that's a bug or if it's just the game being too "sensitive".

**Comment by jvook** [comment]
I don't have the hardware to test whether this issue occurs on real PSP

**Comment by unknownbrackets** [comment]
Repeat rate is managed by the PSP game, so it's not like PSP has a setting or is sending too many presses to the game.

That said, I just double checked to compare with a real PSP, since my memory was foggy.  This happens on a real PSP, but it doesn't happen nearly as readily.  It seems like if you hold up or down, the cursor seems to move close to twice as fast in PPSSPP.  That said, this part isn't related to libretro.

I wonder if it's supposed to actually be 30 FPS there or something.

-[Unknown]

**Comment by bslenul** [comment]
> I wonder if it's supposed to actually be 30 FPS there or something.

Yeah, just watched a couple videos on youtube, it's super obvious when you look at the butterfly wing beats, they are muuuuuch slower on the real PSP or on PSTV, example: https://youtu.be/_AI2Wr2yFgM?t=117

And now someone that is probably using PPSSPP: https://youtu.be/fMtnJYIQf5c?t=106

**Comment by jvook** [comment]
It does the same behavior in game with camera rotation
